### PR TITLE
style: update type ascription to new syntax

### PR DESCRIPTION
**Pull request by jaschdoc**
Added parentheses to type ascription.

Closes https://github.com/flix/flix/issues/7304

### CODE REVIEW

**Comment by jaschdoc** [comment]
There are quite a few tests that use ascription without parenthesis. Maybe @herluf-ba you can ping me once you have implemented the requirement in the new parser and I can submit a PR to your branch?

**Comment by magnus-madsen** [comment]
> There are quite a few tests that use ascription without parenthesis. Maybe @herluf-ba you can ping me once you have implemented the requirement in the new parser and I can submit a PR to your branch?

We should be able to do the refactor now, right?

**Comment by magnus-madsen** [comment]
What I mean is: Lets get it done now. Then Herluf can rebase. We don't want to wait around for each other.

File: main/src/library/Fixpoint/Ast/Shared.flix [hunk 1]
```diff
@@ -54,10 +54,10 @@ mod Fixpoint.Ast.Shared {
     pub def lattice(): Denotation[t] with LowerBound[t], JoinLattice[t], MeetLattice[t] =
         // TODO: Why are these type annotations necessary?
         Denotation.Latticenal(
-            LowerBound.minValue() : t,
-            (x: t, y: t) -> PartialOrder.lessEqual(x: t, y: t),
-            (x: t, y: t) -> JoinLattice.leastUpperBound(x: t, y: t),
-            (x: t, y: t) -> MeetLattice.greatestLowerBound(x: t, y:t)
+            (LowerBound.minValue() : t),
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Ast/Shared.flix:57]: I suspect these annotations can be dropped. Can you check in next PR? Thanks!

File: main/src/library/GetOpt.flix [hunk 2]
```diff
@@ -301,7 +301,7 @@ mod GetOpt {
     def findOptionByName(optname: String, optDescrs: List[OptionDescr[a]]): Result[GetOptionFailure[a], OptionDescr[a]] =
         let getWith     = p -> List.filterMap(o -> { let xs = o.optionNames; if (List.exists(p(optname), xs)) Some(o) else None}, optDescrs);
         // Exact match of `optname` ...
-        let exact       = getWith((x,y) -> x == y): List[OptionDescr[a]];
+        let exact       = (getWith((x,y) -> x == y): List[OptionDescr[a]]);
```
Comment (magnus-madsen) [inline main/src/library/GetOpt.flix:304]: I also suspect this can be dropped.

File: main/src/library/Fixpoint/Interpreter.flix [hunk 3]
```diff
@@ -84,7 +84,7 @@ mod Fixpoint.Interpreter {
                 }, MutMap.getWithDefault(ramSym, MutMap.new(rc1), db))
             case RelOp.Functional(RowVar.Index(i), f, terms, body) =>
                 let args = terms |> Vector.map(evalTerm(env));
-                let result = f(args): Vector[Vector[v]];
+                let result = (f(args): Vector[Vector[v]]);
```
Comment (magnus-madsen) [inline main/src/library/Fixpoint/Interpreter.flix:87]: and this can be dropped.

**Review by magnus-madsen — APPROVED** [review]

**Comment by herluf-ba** [comment]
@jaschdoc Sorry for being slow on this 😅
Here is a list of all the test files that contain a type-ascription that still needs refactoring:
```
Test.Exp.Force.flix
Test.Equality.Set.flix
Test.Exp.ForEachYield.flix
Test.Exp.Ref.Assign.flix
Test.Term.Lit.Set.flix
Test.Equality.Map.flix
Test.Term.Lit.Option.flix
Test.Exp.Ref.Ref.flix
Test.Exp.Match.List.flix
Test.Dec.TypeAlias.flix
Test.Equality.Tag.flix
Test.Derives.Order.flix
Test.Derives.Eq.flix
Test.Predicate.Guard.flix
Test.Kind.Trait.flix
Test.Derives.Hash.flix
Test.Predicate.Functional.flix
Test.Exp.Interpolation.flix
Test.Term.Lit.Result.flix
Test.Term.Lit.List.flix
```

**Comment by jaschdoc** [comment]
Thanks! On it

**Pull request merged**